joomla/joomla-cms
# MSSQL fix sql syntax in  administrator/components/com_admin/sql/updates/sqlazure

### Summary of Changes

Update sql files for mssql database has invalid syntax.
This PR fix files from 3.5.0 and one older to correct syntax.


### Testing Instructions
* Install Joomla 3.5.0 on sqlsrv (mssql) database in folder `joomla`
* Download and extract Joomla from `https://github.com/csthomas/joomla-cms/archive/msfixupdate.zip` to folder `joomla37`
* Move `joomla` to `joomla_old`
* Move `joomla37` to `joomla`
* Copy `joomla_old/configuration.php` to `joomla/configuration.php`
* Download `postupdate.php` file from `https://gist.github.com/mbabker/d7bfb4e1e2fbc6b7815a733607f89281#file-postupdate-php` and save it in `joomla/administrator/` folder.
* [optional] Add a few changes as on below diff file to debugging if you like:
```diff
diff --git a/libraries/joomla/database/driver/sqlsrv.php b/libraries/joomla/database/driver/sqlsrv.php
index a0b4285cfb..d2a2097a3d 100644
--- a/libraries/joomla/database/driver/sqlsrv.php
+++ b/libraries/joomla/database/driver/sqlsrv.php
@@ -630,7 +630,7 @@ class JDatabaseDriverSqlsrv extends JDatabaseDriver
                {
                        $array = array();
                }
-
+echo "$query\n\n";
                // Execute the query. Error suppression is used here to prevent warnings/notices that the connection has been lost.
                $this->cursor = @sqlsrv_query($this->connection, $query, array(), $array);
 
@@ -685,7 +685,7 @@ class JDatabaseDriverSqlsrv extends JDatabaseDriver
                        {
                                // Get the error number and message from before we tried to reconnect.
                                $this->errorNum = $errorNum;
-                               $this->errorMsg = $errorMsg;
+                               $this->errorMsg = $errorMsg;echo "\n\n $errorMsg\n\n";
 
                                // Throw the normal query exception.
                                JLog::add(JText::sprintf('JLIB_DATABASE_QUERY_FAILED', $this->errorNum, $this->errorMsg), JLog::ERROR, 'database-error');
```
* run `php -f administrator/postupdate.php`
* From time to time if you apply diff file to see debug info you may see:
```
DELETE 
FROM [j37_session]
WHERE [time] < N'1487235471'

Error displaying the error page: Application Instantiation Error: Failed to start the session because headers have already been sent by ".../libraries/joomla/database/driver/sqlsrv.php" at line 633.
```
Then try again run `php` file, after a few times it won't failed:) 
* At the end of logs you should see:
```
Update to 3.7.0-beta3 completed successfully.
```
or if this PR is still not perfect then (TEST FAILED):
```
Failed to finalize the upgrade, please check the logs for additional details.
```

Alternative way to test for git users:
```
git clone ...
git checkout -f 3.5.0
rm configuration.php # if exists
# Install Joomla 3.5.0 on sqlsrv
git checkout -f staging
curl https://patch-diff.githubusercontent.com/raw/joomla/joomla-cms/pull/14102.diff | git apply
# Apply above diff file
php -f administrator/postupdate.php
...
```

**I have tested only from Joomla 3.5.0 because I have not php 5.6 or less.**

### Expected result
Upgrade from Joomla 3.5.0 to 3.7 will go without errors.


### Actual result
Upgrade fails.


### Documentation Changes Required
None
